### ISSUE TITLE
Challenges in reproducing Premier_TACO results for certain tasks

### ISSUE THREAD

**Issue by SaintExup3ry** [PremierTACO/premier-taco#2]
![image](https://github.com/PremierTACO/premier-taco/assets/112502788/b6142394-c70a-4760-b0a2-6023cac39917)

I download dataset and try to load the pretrained encoder for downstream few-shot behavior cloning for evaluation, with this line `python train_bc.py task_name=${TASK} seed=${SEED} exp_name=${BC_EXP_NAME} encoder_dir=${ENCODER_CKPT}  offline_data_dir=${DATA_DIR}/dmc_eval_data/${TASK} &`.

However, I have encountered difficulties in reproducing the results reported in the original Premier_TACO paper for several tasks, as evidenced by the significant differences between the 'normalized_reward'(expert relative score) and 'paper_reward' columns in the provided table. 

Tasks with substantial discrepancies include: 'finger_spin', 'hopper_hop', 'cheetah_run', 'walker_walk', and 'humanoid_walk'

Despite following the methodology and setup described in the paper, the model's performance on these tasks deviates considerably from the expected results. I have double-checked the implementation, hyperparameters, and environment configurations, but the issue persists.

I would appreciate any insights or suggestions from the community regarding potential causes or steps to troubleshoot and address these reproducibility challenges. Please let me know if you require any additional information or clarification.

**Comment by FrankZheng2022** [comment]
Hi, this is weird because in all of our previous experiments, we have not experienced this low reward for hopper hop and finger spin. What pretained encoder do you load for the evaluation?

On our side, we have evaluated all the downstream tasks with our provided pretrained visual encoder again using our evaluation trajectories and the result is the following. (Results are averaged over 4 random seeds, not a single seed.)
|  Tasks | Reward | Normalized Reward |
| ----------- | -------| --------|
|Humanoid Walk (100-shot) | 470.8 |58.1|
|Dog Trot (100-shot) | 677.6 |94.1| 
|Hopper Hop (20-shot)|483.1 |84.9|
|Walker Walk (20-shot)|759.1 |79.3|
|Finger Spin (20-shot)|721.0 |75.1|
|Cup Catch (20-shot)|964.6 |99.7|
|Quadruped Walk (20-shot)|688.4 |83.3|
|Quadruped Run (20-shot)|530.4 |70.1|
|Cheetah Run (20-shot)|581.3 |63.8|

**Comment by FrankZheng2022** [comment]
Also we noticed that in our eval config file, the number of time steps (``num_train_steps``) should be set to ```100,000``` instead of ```50,000```, although we don't think this should cause this big number difference. (The numbers in the table above are reproduced by the current number of gradient steps ```50,000```.)

**Comment by SaintExup3ry** [comment]
I think there is a difference in dependencies. Can you tell me the version of pytorch, dm_conrol, and mujoco? Some are not in conda_env.yaml

**Comment by FrankZheng2022** [comment]
Hi, here is the version of my PyTorch, dm_control, and mujoco-py. Happy to help you reproduce our results!
```
torch: 2.2.0+cu118
mujoco-py: 2.1.2.14 (mujoco 210 installed)
dm_control: 1.0.13
```

**Comment by SaintExup3ry** [comment]
I'm working on reproducing the results of the Premier TACO project, and I have a question regarding the bc_config file. In the provided configuration file, I couldn't find the freeze_encoder flag. Should I assume that the encoder should be trainable and set freeze_encoder to False?
And I saw there's mw library in train_bc.py, but it does not used. If the mw library is not essential for reproducing the core results, I would prefer to avoid changing my numpy version, as it may introduce compatibility issues with other dependencies. Is it necessary to install mw library?

Thank you for your assistance in ensuring I follow the correct setup for reproducing the results accurately.

**Comment by FrankZheng2022** [comment]
Yes, you are right that ``freeze_encoder`` should be set to False. (It is due to a mismatch between our local codebase and the official GitHub codebase and we have fixed this issue.) For ``import mw``, yes you don't need meta world to be installed, it was only used for Meta-World experiments.
If you have any further questions, please let us know.
